### PR TITLE
added second volume to Valheim pod to persist server installation on pod recreation

### PR DESCRIPTION
**Pull request by fbuchmeier**
Hi @Addyvan,

I've added a second optional volume to the Valheim deployment so that the server installation files don't have to be downloaded every time a new pod is created.

References for this can be found here: https://github.com/lloesche/valheim-server-docker#basic-docker-usage

To not break any existing deployments, this is a seperate root variable (`serverStorage`) instead of an update to the already existing `storage`.

Cheers,
Florian.

### CODE REVIEW

File: chart/templates/service.yaml [hunk 1]
```diff
@@ -8,11 +8,17 @@ spec:
     port: {{ .Values.networking.gamePort | int }}
     targetPort: {{ .Values.networking.gamePort | int }}
     protocol: UDP
+    {{- if eq .Values.networking.serviceType "NodePort" }}
```
Comment (Addyvan) [inline chart/templates/service.yaml:11]: This seems unrelated to the storage, can you elaborate more on this change please?

**Review by Addyvan — APPROVED** [review]
lgtm. Will test it out before releasing a new chart version.

File: chart/templates/service.yaml (hunk 1 above)
Comment (fbuchmeier) [reply chart/templates/service.yaml:11]: Hey, sorry for that, I pushed the changes back to the same branch, actually intended to open a second PR for this as you suggested.

This enables setting a specific NodePort.

I am running your chart on bare metal servers without a loadbalancer, so a fixed port (23456) was much nicer for me than an automatically assigned one.

This however has the downside that the user of the chart  has to make sure this port is not already in use by another service (inside or outside of Kubernetes).

Comment (Addyvan) [reply chart/templates/service.yaml:11]: Makes sense, no harm no foul. Merging.
